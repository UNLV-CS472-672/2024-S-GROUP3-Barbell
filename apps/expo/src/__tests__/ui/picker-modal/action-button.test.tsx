import React from 'react'
import { TouchableOpacity, TouchableOpacityProps } from 'react-native'

import { fireEvent, render } from '@testing-library/react-native'

import ActionButton from '~/components/ui/picker-modal/action-button/action-button'

interface MockTouchableProps extends TouchableOpacityProps {
  testID?: string
}

const MockTouchable: React.FC<MockTouchableProps> = ({ children, onPress, testID, ...props }) => (
  <TouchableOpacity onPress={onPress} testID={testID} {...props}>
    {children}
  </TouchableOpacity>
)

describe('ActionButton', () => {
  it('renders correctly with text', () => {
    const { getByText } = render(
      <ActionButton text='thienguen' TouchableComponent={MockTouchable} />,
    )
    expect(getByText('thienguen')).toBeTruthy()
  })

  it('handles press events', () => {
    const onPressMock = jest.fn()
    const { getByTestId } = render(
      <ActionButton onActionPress={onPressMock} TouchableComponent={MockTouchable} />,
    )
    fireEvent.press(getByTestId('action-button'))
    expect(onPressMock).toHaveBeenCalled()
  })

  it('conditionally renders the Divider', () => {
    const { queryByTestId } = render(
      <ActionButton isLastItem={true} TouchableComponent={MockTouchable} />,
    )
    // Since the `queryByTestId` for 'divider' would not work because the Divider itself does not accept testID as a prop,
    expect(queryByTestId('divider')).toBeNull()
  })
})
