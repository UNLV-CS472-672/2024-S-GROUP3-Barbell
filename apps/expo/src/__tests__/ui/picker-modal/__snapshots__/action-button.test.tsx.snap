// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton matches snapshot 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderBottomLeftRadius": 0,
        "borderBottomRightRadius": 0,
        "height": 75,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="action-button"
  >
    <Text
      className="font-koulen"
      style={
        [
          {
            "color": "#fff",
          },
          undefined,
        ]
      }
    >
      Snapshot Test
    </Text>
  </View>,
  <View
    style={
      [
        {
          "backgroundColor": "#3b3b3b",
          "height": 1,
          "width": "100%",
        },
        undefined,
      ]
    }
    testID="divider"
  />,
]
`;
